### PR TITLE
mgr/orch,cephadm: add timestamps to daemons and services

### PR DESCRIPTION
**Pull request by liewegas**
End result is:
```
gnit:build (cephadm-btime) 11:49 AM $ bin/ceph orch ls
NAME     RUNNING  REFRESHED  AGE  SPEC     PLACEMENT  IMAGE NAME                                      IMAGE ID      
crash        1/1  97s age    28m  present  all=true   docker.io/ceph/daemon-base:latest-master-devel  7a8f71d9429a  
mds.foo      1/1  97s age    24m  present  count=1    docker.io/ceph/daemon-base:latest-master-devel  7a8f71d9429a  
gnit:build (cephadm-btime) 11:49 AM $ bin/ceph orch ps
NAME                 HOST  STATUS         REFRESHED  AGE  VERSION               IMAGE NAME                                      IMAGE ID      CONTAINER ID  
crash.gnit           gnit  running (28m)  99s ago    28m  15.1.0-1433-gf2f7381  docker.io/ceph/daemon-base:latest-master-devel  7a8f71d9429a  fa9286078673  
mds.foo.gnit.dgtpua  gnit  running (28m)  99s ago    28m  15.1.0-1433-gf2f7381  docker.io/ceph/daemon-base:latest-master-devel  7a8f71d9429a  0b42bd574d69  
```

### CODE REVIEW

**Comment by liewegas** [comment]
retest this please

**Comment by mgfritch** [comment]
jenkins test make check

**Comment by liewegas** [comment]
pass:
http://pulpito.ceph.com/sage-2020-03-05_01:22:18-rados:cephadm-wip-sage3-testing-2020-03-04-1654-distro-basic-smithi/
http://pulpito.ceph.com/sage-2020-03-05_01:19:52-rados:cephadm-wip-sage3-testing-2020-03-04-1654-distro-basic-smithi/

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -608,6 +611,53 @@ def pathify(p):
         return os.path.join(os.getcwd(), p)
     return p
 
+def get_file_timestamp(fn):
+    try:
+        mt = os.path.getmtime(fn)
```
Comment (mgfritch) [inline src/cephadm/cephadm]: I somewhat wonder if ctime would be more robust ... ?

File: src/pybind/mgr/cephadm/module.py [hunk 2]
```diff
@@ -131,14 +131,18 @@ def __init__(self, mgr):
...
+        self.spec_created = {} # type: Dict[str, datetime.datetime]
 
     def load(self):
         # type: () -> None
         for k, v in six.iteritems(self.mgr.get_store_prefix(SPEC_STORE_PREFIX)):
             service_name = k[len(SPEC_STORE_PREFIX):]
             try:
-                spec = ServiceSpec.from_json(json.loads(v))
+                v = json.loads(v)
+                spec = ServiceSpec.from_json(v['spec'])
```
Comment (jschmid1) [inline src/pybind/mgr/cephadm/module.py]: We actually want the spec _and_ the placement. -> `spec = ServiceSpec.from_json(v)`

Comment (jschmid1) [reply src/pybind/mgr/cephadm/module.py]: oh, nvm.. I should've read the new `save()` method

File: src/pybind/mgr/orchestrator/module.py [hunk 3]
```diff
@@ -335,30 +335,36 @@ def ukn(s):
...
             table.align['SPEC'] = 'l'
             table.align['PLACEMENT'] = 'l'
             table.left_padding_width = 0
             table.right_padding_width = 2
             for s in sorted(services, key=lambda s: s.service_name):
-                if s.last_refresh:
-                    age = to_pretty_timedelta(now - s.last_refresh) + ' ago'
-                else:
-                    age = '-'
+                def nice_delta(t, suffix=''):
```
Comment (jschmid1) [inline src/pybind/mgr/orchestrator/module.py]: maybe factor that out as this appears twice

**Pull request merged**